File: java/src/main/java/com/techelevator/service/AccountService.java
package com.techelevator.service;

import com.techelevator.dao.AccountDao;
import com.techelevator.dao.GameDao;
import com.techelevator.dao.UserDao;
import com.techelevator.model.Account;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Component;
import org.springframework.web.server.ResponseStatusException;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Component
public class AccountService {

    private AccountDao accountDao;
    private UserDao userDao;
    private GameDao gameDao;

    public AccountService(AccountDao accountDao, UserDao userDao, GameDao gameDao) {
        this.accountDao = accountDao;
        this.userDao = userDao;
        this.gameDao = gameDao;
    }

    public Account getAccountByAccountId(int id) {
        return accountDao.getAccountByAccountId(id);
    }

    public List<Account> getCurrentUserAccounts(String username) {
        return accountDao.getAccountsByUserId(userDao.findIdByUsername(username));
    }

    public List<Account> getAccountsByUserId(int id) {
        return accountDao.getAccountsByUserId(id);
    }

    public List<Account> getAccountsByGameId(int id) {
        return accountDao.getAccountsByGameId(id);
    }

    public List<Account> createAccountsByGameId(int gameId, Integer[] userIds) {
        List<Account> accounts = new ArrayList<>();
        for (int id : userIds) {
            Account account = accountDao.createAccount(id, gameId);
            if (account == null) {
                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid username");
            }
            accounts.add(account);
        }
        return accounts;
    }

    public Account createAccount(int userId, int gameId) {
        List<Account> accounts = accountDao.getAccountsByGameId(gameId);
        for (Account account : accounts) {
            if (account.getUserId() == userId) {
                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User is already part of the game.");
            }
        }
        return accountDao.createAccount(userId, gameId);
    }

    public void updateBalance(int id, BigDecimal newBalance) {
        accountDao.updateBalance(id, newBalance);
    }
}

File: vue/src/services/AccountService.js
import axios from "axios";

export default {

    getAccountByAccountId(id) {
        return axios.get(`/accounts/${id}`, id);
    },

    getAccountByUserId(userId) {
        return axios.get(`/accounts/${userId}/accounts`, userId);

    },

    getCurrentUserAccounts() {
        return axios.get('/users/me/accounts');
    },

    getAccountsByGameId(gameId) {
        return axios.get(`/games/${gameId}/accounts`);

     },
     
    createAccountsByGameId(gameId, userIds) {
        return axios.post(`/accounts/${gameId}/create`, userIds);
    },

    createAccount(userId, gameId) {
        return axios.post('/accounts/create', userId, gameId);

     },
    
     updateBalance(id) {
         return axios.put(`/accounts/${id}/balance`) // might need entire object to update balance (account.id)

    },

    getAccountSummary(id) {
        return axios.get(`/accounts/${id}/costBasis`)
    }

    // newBalance() {

    //  },

}
